### PR TITLE
Adds CODEOWNERS file to restrict review to product team

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

medic/cht-core#6258

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
@kennsippell Can I get a review for this? Once this is merged I think I can open up the write permissions to more people in the org and reviewing should be limited to the product team.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+*       @medic/product-team
```
Comment (kennsippell) [inline .github/CODEOWNERS:1]: FYI this team includes Derick, Jacqueline, QAs, SREs, etc. https://github.com/orgs/medic/teams/product-team/members
Do you want them signing off on PRs?
Unsure if you want https://github.com/orgs/medic/teams/development-team/members?

Comment (garethbowen) [reply .github/CODEOWNERS:1]: Yeah I guess. The plan was to work on redefining the teams a little, but development is probably closer to what we want.

**Comment by garethbowen** [comment]
> Consider granting @medic/app-services-team permission to sign-off on code changes only affecting config folder?

We could do, though I think it's important that a dev team member knows what's going on in the config directory. In this case the code review would be more about knowledge sharing than gatekeeping. Does that make sense?

**Comment by kennsippell** [comment]
@garethbowen Yes, that makes sense to me. Standard I think is a litlte more clear cut, I know you have a dependency on default. So app-services always adds an "FYI reviewer" from the product team on a config PR. Is that what you have in mind?

**Comment by garethbowen** [comment]
For the reference app and default config, yes.

**Pull request merged**